thavalon-webapp/src/components/gameComponents/playerBoard.tsx

```diff
@@ -1,5 +1,177 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { GameSocket, InboundMessage, InboundMessageType, OutboundMessageType } from "../../utils/GameSocket";
+import { Vote, GameMessageType, GameMessage, Snapshot } from "./constants";
+import { Spinner } from "react-bootstrap";
 
+import "../../styles/gameStyles/playerBoard.scss";
+
+
+/**
+ * Props object for the PlayerCard component. Some of these aren't used yet,
+ * pending future development.
+ */
+interface PlayerCardProps {
+    name: string
+    toggleSelected: (name: string) => void,
+    me?: boolean,
+    tabbedOut?: boolean,
+    isProposing?: boolean,
+    isSelected?: boolean,
+    vote?: Vote,
+    declaredAs?: string
+}
+
+/**
+ * Message for the tabbed out indicator message.
+ */
+interface PlayerFocusChangeMessage {
+    displayName: string,
+    isTabbedOut: boolean
+}
+
+/**
+ * Board containing the player list and all related functions. This is a container
+ * for all PlayerCards and has handlers for game and lobby events.
+ */
 export function PlayerBoard(): JSX.Element {
-    return <h1>Player Board</h1>;
+    // On mount, set up event handlers for the game socket and the DOM.
+    // On unmount, clean up event handlers.
+    useEffect(() => {
+        const connection = GameSocket.getInstance();
+        connection.onGameEvent.subscribe(handleMessage);
+        connection.onLobbyEvent.subscribe(handleMessage);
+        document.onvisibilitychange = () => sendPlayerVisibilityChange();
```

Comment:
Would something like https://blog.sethcorker.com/harnessing-the-page-visibility-api-with-react be better and more react-y than directly modifying document.onvisibilitychange?

Reply:
addEventHandler is, I believe, the old method for dealing with onVisibilityChange. onVisibilityChange was added so that users could directly subscribe to the event, I believe. Everything else in that article appears to be support for other browsers, which I'm not too concerned with right now.